content/doc/developer/publishing/documentation.adoc

```diff
@@ -0,0 +1,169 @@
+---
+title: Plugin Documentation
+layout: developersection
+---
+
+Documentation is an important part of any Jenkins plugin.
+It includes user documentation (plugin pages, changelogs, user guidelines, etc.)
+and the contributor documentation (how to contribute, developer guidelines, etc.).
+This section provides overview of these documentation types.
+
+== Plugin pages
+
+Plugin pages are hosted on the link:https://plugins.jenkins.io/[Plugin Site],
+these pages are being generated automatically using the metadata from the latest plugin release and an external documentation page.
+External documentation can be retrieved from GitHub or from https://wiki.jenkins.io[Jenkins Wiki].
+Since the introduction of GitHub documentation support in September 2019
+(https://groups.google.com/forum/#!topic/jenkinsci-dev/VSdfVMDIW-A[announcement]),
+this is the *recommended way* to host documentation.
+
+In Jenkins we recommend storing documentation in the plugin repository.
+It allows plugin maintainers to provide the same documentation from GitHub README pages and the Jenkins plugin site,
+and at the same time it allows using the Documentation-as-Code techniques when the documentation is a part of the
+repository and hence all common practices can be applied 
+(versioning, pull requests and reviews, creating documentation in parallel with features, editing docs from GitHub Web UI).
+
+=== Using GitHub as a source of documentation
+
+Since the introduction of GitHub documentation support in September 2019
+(https://groups.google.com/forum/#!topic/jenkinsci-dev/VSdfVMDIW-A[announcement]),
+this is the *recommended way* to host documentation. It allows plugin
+maintainers to provide the same documentation from GitHub pages and the
+Jenkins plugin site, and at the same time it allows using the
+Documentation-as-Code techniques when the documentation is a part of the
+repository and hence all common practices can be applied (versioning,
+pull requests and reviews, creating documentation in parallel with
+features, editing docs from GitHub Web UI).
+
+The plugin site can pull documentation from the root README pages and
+from other locations. For new plugins it is recommended to use README
+pages as a source of documentation, preferably in Markdown or Asciidoc
+format (TXT is also supported in such case).
+
+How to publish the plugin documentation from GitHub?
+
+* Create a README page and put the plugin documentation there
+** This page will become a landing page for https://plugins.jenkins.io/
```

Comment:
Weird nesting of items looks like this was transplanted from a really bad Powerpoint presentation.